prepare_check_cleanup/migration.py

```diff
@@ -0,0 +1,269 @@
+        API to the GitHub repository.
+    """
+    repository = create_repository_client(access_token=github_access_token, base_path=Path())
+    # Accessing private client since this is for testing pourposes only, otherwise would need to
+    # add code to pull_request only needed for testing
+    return repository._github_repo  # pylint: disable=protected-access
+
+
+def _get_migration_pull_request(github_repo: Repository) -> PullRequest | None:
+    """Get the migration pull request if it exists.
+
+    Args:
+        github_repo: API for interacting with GitHub.
+
+    Returns:
+        The migration pull request or None if it doesn't exist.
+    """
+    return next(
+        (
+            pull_request
+            for pull_request in github_repo.get_pulls()
+            if pull_request.title == ACTIONS_PULL_REQUEST_TITLE
+        ),
+        None,
+    )
+
+
+def _get_migration_branch(github_repo: Repository) -> GitRef | None:
+    """Get the migration branch if it exists.
+
+    Args:
+        github_repo: API for interacting with GitHub.
+
+    Returns:
+        The migration branch or None if it doesn't exist.
+    """
+    with suppress(GithubException):
+        return github_repo.get_git_ref(f"heads/{DEFAULT_BRANCH_NAME}")
+    return None
+
+
+def check_branch(github_access_token: str) -> bool:
+    """Check that the migration branch was created.
+
+    Args:
+        github_access_token: The secret required for interactions with GitHub.
+
+    Returns:
+        Whether the test succeeded.
+    """
+    test_name = "check-branch"
+
+    github_repo = _create_repository_client(github_access_token=github_access_token)
+    migration_branch = _get_migration_branch(github_repo=github_repo)
+    if not migration_branch:
+        logging.error(
+            "%s check failed, migration branch %s not created", test_name, DEFAULT_BRANCH_NAME
+        )
+        return False
+
+    logging.info("%s check succeeded", test_name)
+    return True
+
+
+def check_pull_request(github_access_token: str) -> bool:
+    """Check that the migration pull request was created.
+
+    Args:
+        github_access_token: The secret required for interactions with GitHub.
+
+    Returns:
+        Whether the test succeeded.
+    """
+    test_name = "check-pull-request"
+
+    github_repo = _create_repository_client(github_access_token=github_access_token)
+    migration_pull_request = _get_migration_pull_request(github_repo=github_repo)
+    if not migration_pull_request:
+        logging.error(
+            "%s check failed, migration pull request %s not created",
+            test_name,
+            ACTIONS_PULL_REQUEST_TITLE,
+        )
+        return False
+
+    expected_files = {
+        f"{DOCUMENTATION_FOLDER_NAME}/{DOCUMENTATION_INDEX_FILENAME}",
+        f"{DOCUMENTATION_FOLDER_NAME}/page.md",
+    }
+    files_in_pull_request = {file.filename for file in migration_pull_request.get_files()}
+    if expected_files == files_in_pull_request:
```

Comment:
Shouldn't it be `!=` ?

Reply:
Yes, thanks for spotting this, I think there must still be a bug in the migration e2e tests then